src/dataflow/src/sink/kafka.rs

```diff
@@ -242,7 +244,52 @@ pub fn kafka<G>(
 where
     G: Scope<Timestamp = Timestamp>,
 {
+    let name = format!("kafka-{}", id);
+
     let stream = &collection.inner;
+
+    let encoder = Encoder::new(key_desc, value_desc, connector.consistency.is_some());
+    let key_schema_id = connector.key_schema_id;
+    let value_schema_id = connector.value_schema_id;
+
+    let encoded_stream = avro_encode_stream(
+        stream,
+        as_of.clone(),
+        connector
+            .consistency
+            .clone()
+            .and_then(|consistency| consistency.gate_ts),
+        encoder,
+        key_schema_id,
+        value_schema_id,
+        connector.fuel,
+        name.clone(),
+    );
+
+    produce_to_kafka(encoded_stream, id, name, connector, as_of)
+}
+
+/// Produces/sends a stream of encoded rows (as `Vec<u8>`) to Kafka.
+///
+/// This operator exchanges all updates to a single worker by hashing on the given sink `id`.
+///
+/// Updates are only sent to Kafka once the input frontier has passed their `time`. Updates are
+/// sent in ascending timestamp order. The order of updates at the same timstamp will not be changed.
+/// However, it is important to keep in mind that this operator exchanges updates so if the input
+/// stream is sharded updates will likely arrive at this operator in some non-deterministic order.
+///
+/// Updates that are not beyond the given [`SinkAsOf`] and/or the `gate_ts` in
+/// [`KafkaSinkConnector`] will be discarded without producing them.
```

Comment:
This is a slightly different use of `as_of` than we use elsewhere in the code base. Idk if it is new here or something being borrowed, but: the conventional use of `as_of` is to indicate that all times should be advanced to the frontier described by `as_of`. Whether such updates should be discarded or not (vs advanced) is traditionally a bool `snapshot`.

It is possible that this isn't a new issue in sink code; if it isn't new we should file an issue to straighten out the language.

Reply:
It's a new-ish thing, added right before my time: https://github.com/MaterializeInc/materialize/commit/06c0f76257d7582793540906c83a89e681d5194d